luarules/gadgets/unit_xp_range_bonus.lua

```diff
@@ -57,8 +63,8 @@ function gadget:UnitDamaged(unitID, unitDefID, unitTeam, damage, paralyzer, weap
 		return
```

Comment:
Can you tell me why the check above is necessary?

Reply:
Not entirely sure, I believe it was due to some bugs but I will try it without it.

Reply:
You can do it like this to log a possible problematic case (inside the if):

`Spring.Echo(os.clock(), 'attackerDefID not well defined!', attackerDefID)`

Then test the case and see what it spits

Reply:
attackerDefID can be ground, lava, collision etc

Reply:
A string?

Reply:
no, numbers like -1 or -2 which stand for those types of damage

Reply:
I see, so the check can be replaced with ` > 0`

Reply:
Removed it, the issue doesn't seem to be happening any more as I have done a small bit of testing.

Reply:
Shouldn't it be just `nil` instead as per [source](https://github.com/beyond-all-reason/spring/blob/337da94e0c6b48ae4ccedf924aa84692a6301b01/rts/Lua/LuaHandle.cpp#L1223-L1228)?